### DISCUSSION TITLE
Can't compile because of undefined types, did you mean "xxx?"

### DISCUSSION THREAD

**Discussion by Fabxx** [ocornut/imgui#5347] [New Users Build/Link/Run issues ONLY!]
Here's my compile log:
[log.txt](https://github.com/ocornut/imgui/files/8770051/log.txt)

the include order is:
```#include <iostream>
#include <GLFW/glfw3.h>
#include "imgui.h"
#include "imgui_impl_opengl3_loader.h"
#include "imgui_impl_opengl3.h"

int main() 
{
	int counter;
	 if (ImGui::Button("Click me"))
        ++counter;
    ImGui::Text("counter=%i", counter);
}
```

The GL version i have is 4.6.0, also the glad files and GLFW are based on the lastes version.

**Comment by PathogenDavid** [comment]
As [noted at the top of the file](https://github.com/ocornut/imgui/blob/89a28209e89f69a5eda560d2ef9c13915c95616a/backends/imgui_impl_opengl3_loader.h#L9-L10), you shouldn't include `imgui_impl_opengl3_loader.h` directly.

Also even if your project built it would not work at all. See [`example_glfw_opengl3`](https://github.com/ocornut/imgui/blob/5139fb7e1869254dc9753e7b07a734823cc1ba5d/examples/example_glfw_opengl3/main.cpp) for details on using Dear ImGui with OpenGL and GLFW.

Edit: I missed that your log is showing errors from `imgui_impl_opengl3.cpp`. Those errors don't match up with the current version of `imgui_impl_opengl3_loader.h`, what commit of Dear ImGui are you using?

**Comment by Fabxx** [comment]
as for the project yeah it wouldn't work, i needed to check those errors first. as for the dear imgui commit i just downloaded the updated source today. I just retried now by replacing again the files and following the includes in the guide sended, but this time i get this log:
[log.txt](https://github.com/ocornut/imgui/files/8771772/log.txt)

**Comment by Fabxx** [comment]
Update: i've been using in vs code the c/c++ project generator with simple makefile to make things easier to setup, and i can indicate the directories just fine, but the problem is still in the linker with the objects not found:
[log.txt](https://github.com/ocornut/imgui/files/8774577/log.txt)

makefile:
[makefile.txt](https://github.com/ocornut/imgui/files/8777345/makefile.txt)



Screenshot of the directories tree:
![image](https://user-images.githubusercontent.com/30447649/170368642-221c4f7a-2ebb-4ece-96b2-60f54f8acdc2.png)

**Comment by Fabxx** [comment]
I've fixed the issue by adding another SRC2 parameter with the sources of imgui, and then making the compiler compile the cpp objects of imgui, so now i can use the funcitons
